egs/babel_multilang/s5/local/nnet3/run_common_langs.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+. ./cmd.sh
+set -e
+stage=1
+train_stage=-10
+generate_alignments=true # false if doing ctc training
+speed_perturb=true
+use_flp=false
+use_pitch=true
+pitch_conf=conf/pitch.conf
+voicing_conf=
+use_pitch_plp=false
+[ ! -f ./conf/common_vars.sh ] && echo 'the file conf/common_vars.sh does not exist!' && exit 1
+
+. conf/common_vars.sh || exit 1;
+
+[ -f local.conf ] && . ./local.conf
+
+. ./utils/parse_options.sh
+
+
+L=$1
+
+# perturbed data preparation
+train_set=train
+if [ "$speed_perturb" == "true" ]; then
+  if [ $stage -le 1 ]; then
+    #Although the nnet will be trained by high resolution data, we still have to perturbe the normal data to get the alignment
+    # _sp stands for speed-perturbed
+    for datadir in train; do
+      utils/perturb_data_dir_speed.sh 0.9 data/$L/${datadir} data/$L/temp1
+      utils/perturb_data_dir_speed.sh 1.1 data/$L/${datadir} data/$L/temp2
+      utils/combine_data.sh data/$L/${datadir}_tmp data/$L/temp1 data/$L/temp2
+      utils/validate_data_dir.sh --no-feats data/$L/${datadir}_tmp
+      rm -r data/$L/temp1 data/$L/temp2
+      
+      featdir=plp_perturbed/$L
+      if $use_pitch_plp; then
+        steps/make_plp_pitch.sh --cmd "$train_cmd" --nj $train_nj  data/$L/${datadir}_tmp exp/$L/make_plp_pitch/${datadir}_tmp $featdir
+      else
+        steps/make_plp.sh --cmd "$train_cmd" --nj $train_nj data/$L/${datadir}_tmp exp/$L/make_plp/${datadir}_tmp $featdir
+      fi
+      steps/compute_cmvn_stats.sh data/$L/${datadir}_tmp exp/$L/make_plp/${datadir}_tmp $featdir || exit 1;
+      utils/fix_data_dir.sh data/$L/${datadir}_tmp
+      
+      utils/copy_data_dir.sh --spk-prefix sp1.0- --utt-prefix sp1.0- data/$L/${datadir} data/$L/temp0
+      utils/combine_data.sh data/$L/${datadir}_sp data/$L/${datadir}_tmp data/$L/temp0
+      utils/fix_data_dir.sh data/$L/${datadir}_sp
+      rm -r data/$L/temp0 data/$L/${datadir}_tmp
+    done
+  fi
+  
+  train_set=train_sp
```

Comment:
When checking for `.done` scripts, print out a statement to the user describing what stage is being skipped, why and what the user can do to force the script to run that stage.

You could probably also check if the `data/*_sp` directories are done and skip the feature extraction.